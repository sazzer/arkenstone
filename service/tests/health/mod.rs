use actix_web::test;
use insta::{assert_display_snapshot, assert_json_snapshot};

#[actix_rt::test]
#[cfg_attr(not(feature = "docker_tests"), ignore)]
async fn test_healthcheck() {
  let service = crate::service::Service::new().await;
  let res = service
    .request(test::TestRequest::get().uri("/health").to_request())
    .await;
  assert_display_snapshot!(res.headers());
  assert_json_snapshot!(res.to_json().unwrap());
}
