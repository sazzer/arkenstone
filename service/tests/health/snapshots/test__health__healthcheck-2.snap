---
source: tests/health/mod.rs
expression: res.to_json().unwrap()
---
{
  "components": {
    "db": {
      "healthy": true
    }
  },
  "healthy": true
}
